dali/operators/reader/nemo_asr_reader_op.h

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_READER_NEMO_ASR_READER_OP_H_
+#define DALI_OPERATORS_READER_NEMO_ASR_READER_OP_H_
+
+#include <fstream>
+#include <map>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+#include <istream>
+#include <memory>
+
+#include "dali/operators/reader/reader_op.h"
+#include "dali/operators/reader/loader/nemo_asr_loader.h"
+
+namespace dali {
+class NemoAsrReader : public DataReader<CPUBackend, AsrSample> {
+ public:
+  explicit NemoAsrReader(const OpSpec& spec):
+      DataReader<CPUBackend, AsrSample>(spec),
+      read_sr_(spec.GetArgument<bool>("read_sample_rate")),
+      read_text_(spec.GetArgument<bool>("read_text")),
+      num_threads_(std::max(1, spec.GetArgument<int>("num_threads"))),
+      thread_pool_(num_threads_, spec.GetArgument<int>("device_id"), false) {
+    loader_ = InitLoader<NemoAsrLoader>(spec);
+  }
+
+  void Prefetch() override {
+    DataReader<CPUBackend, AsrSample>::Prefetch();
+    auto &curr_batch = prefetched_batch_queue_[curr_batch_producer_];
+
+    // Waiting until all the audio samples are ready to be consumed
+    for (auto &sample : curr_batch) {
+      thread_pool_.AddWork([&sample](int tid) {
```

Comment:
Can we infer the priority? Maybe we can use the metadata from JSON if available?